src/v/storage/types.h

```diff
@@ -99,6 +99,11 @@ class snapshotable_stm {
     // log eviction attempts to offsets not greater than this.
     virtual model::offset max_collectible_offset() = 0;
 
+    // Lets the STM limit removal or compaction of data. May refer to a Kafka
```

Comment:
Would it be possible to add a sentence commenting how this is different from `max_collectible_offset` ?

Reply:
Done